Bottom: f97cc0d2ba8c5312baeac56b862395e22e7e3159
Top:    f43f5a81188ce8e4005fc9e3c073c7f70bbf1c65
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-11 17:03:22 +0000

Refresh of speed-up-initial-clipping-pour


---

diff --git a/src/pour.c b/src/pour.c
index cea5f75..240ae0d 100644
--- a/src/pour.c
+++ b/src/pour.c
@@ -671,7 +671,8 @@ ClearPour (DataTypePtr Data, LayerTypePtr Layer, PourType * pour,
       GROUP_LOOP (Data, group);
       {
         r += r_search (layer->line_tree, &region, NULL, line_sub_callback, &info);
-        subtract_poly (info.accumulate, &info.pg);
+        if (info.accumulate != NULL)
+          subtract_poly (info.accumulate, &info.pg);
         info.accumulate = NULL;
         r += r_search (layer->arc_tree,  &region, NULL, arc_sub_callback,  &info);
         r += r_search (layer->text_tree, &region, NULL, text_sub_callback, &info);
@@ -680,7 +681,8 @@ ClearPour (DataTypePtr Data, LayerTypePtr Layer, PourType * pour,
       END_LOOP;
       r += r_search (Data->via_tree, &region, NULL, pin_sub_callback, &info);
       r += r_search (Data->pin_tree, &region, NULL, pin_sub_callback, &info);
-      subtract_poly (info.accumulate, &info.pg);
+      if (info.accumulate != NULL)
+        subtract_poly (info.accumulate, &info.pg);
       info.accumulate = NULL;
     }
